source/MaterialXCore/Document.h

```diff
@@ -566,6 +604,7 @@ class Document : public GraphElement
   private:
     class Cache;
     std::unique_ptr<Cache> _cache;
+    std::unordered_map<string, UnitConverterPtr> _unitConverters;
```

Comment:
I wonder if the registration and retrieval of unit converters should be placed outside the document level, and more globally for the entire API session. So either keeping this map as a singleton registry somewhere (for example as a static member in the UnitConverter base class), or adding a UnitConverterRegistry class that an API user can instantiate and use to keep track of the converters. Then we avoid having to register unit converters on every new document created.

Reply:
The reason why I put this in was to make it convenient to write a "getUnitValue" method, but there are a whole bunch of variants of "getValue" then it seems bad to make this even have more combinations. But agree it's not really a document property. I'll look at having a registry.

Reply:
There is a new UnitConverterRegistry class now.